lib/internal/http2/compat.js

```diff
@@ -188,6 +188,11 @@ function resumeStream(stream) {
 }
 
 const proxySocketHandler = {
+  has(stream, prop) {
+    const ref = stream.session !== undefined ? stream.session[kSocket] : stream;
+    return (prop in stream) || (prop in ref);
```

Comment:
Optional suggestion: The test file only covers one of these conditions, right? Does it make sense to add a test case so that both are covered?